ckanext/s3filestore/uploader.py

```diff
@@ -38,58 +38,23 @@ def get_s3_bucket(self, bucket_name):
         s_key = config.get('ckanext.s3filestore.aws_secret_access_key')
 
         # make s3 connection
-        S3_conn = boto.connect_s3(p_key, s_key)
-
-        # make sure bucket exists and that we can access
-        try:
-            bucket = S3_conn.get_bucket(bucket_name)
-        except boto.exception.S3ResponseError as e:
-            if e.status == 404:
-                log.warning('Bucket {0} could not be found, ' +
-                            'attempting to create it...'.format(bucket_name))
-                try:
-                    bucket = S3_conn.create_bucket(bucket_name)
-                except (boto.exception.S3CreateError,
-                        boto.exception.S3ResponseError) as e:
-                    raise S3FileStoreException(
-                        'Could not create bucket {0}: {1}'.format(bucket_name,
-                                                                  str(e)))
-            elif e.status == 403:
-                raise S3FileStoreException(
-                    'Access to bucket {0} denied'.format(bucket_name))
-            else:
-                raise
-
-        return bucket
+        session = boto3.Session(aws_access_key_id=p_key, aws_secret_access_key=s_key)
+        s3 = session.resource('s3')
+        return s3.Bucket(bucket_name)
 
     def upload_to_key(self, filepath, upload_file, make_public=False):
```

Comment:
I don't see any separation between public and private files (eg group image files should be public and at least the private resources should be made private).

From my initial reading, I'm under the impression that this will depend on the bucket configuration and all uploaded files will have those default permissions. @grahame am I correct?

If so, everything being publicly accessible from the S3 links can be a security issue here.